In [4]:
import numpy as np
import ast
import networkx as nx
import matplotlib.pyplot as plt
from collections import Counter
import matplotlib.cm as cmx
import json
# For color mapping
import matplotlib.colors as colors
from operator import itemgetter

#directorio="D:\\Para procesar\\Académica\\Ciencias Físicas\\Doctorado\\Materias\\Redes complejas\\Trabajos_prácticos\\Practica_computacional\\"#Redes\\"
#directorio='G:\\Redes\\'
path="D:\\Para procesar\\Académica\\Ciencias Físicas\\Doctorado\\Materias\\Redes complejas\\Trabajos_prácticos\\Practica_computacional\\"
version=1

In [7]:
paper

NameError: name 'paper' is not defined

In [21]:
#Filtra papers, saca campos innecesarios y guarda en archivo "papers_filtro_n.txt"
def filtrar_papers(R_n,revista,version):
    paper_conjuntos = []
    
    with open(path+'papers_filtro_'+str(revista)+'_'+str(version)+'.txt','a',encoding='utf-8') as archivo_guardar:
            
        R = open(path+'\mag_papers_'+R_n+'.txt', 'r') 
        N = 10000000
        i = 0
        j=0
        for line in R:
            paper=ast.literal_eval(line)
            try:
                paper['doc_type']
                paper['fos']
                paper['keywords']
                paper['references']
                paper['venue']
                if paper['doc_type']=='Journal' and paper['lang']=='en' and paper['venue'].split()[0]==revista:
                    j+=1
                    paper_conjuntos.append(paper)
                    key='abstract';sacarcampo(paper,key)
                    key='url';sacarcampo(paper,key)
                    key='lang';sacarcampo(paper,key)
                    key='page_start';sacarcampo(paper,key)
                    key='page_end';sacarcampo(paper,key)
                    key='doc_type';sacarcampo(paper,key)
                    key='pdf';sacarcampo(paper,key)
                    json.dump(paper, archivo_guardar, ensure_ascii=False)
                    archivo_guardar.write('\n')
            except KeyError:
                pass
            i+=1
            if i==N:
                break
        R.close()
        print(j)
    archivo_guardar.close()
    return(paper_conjuntos)

#Crea lista de papers filtrados y diccionario de papers citados y su frecuencia
def contador(ides_in,ides_out,paper_conjuntos):
    for line in range(len(paper_conjuntos)):
        ides_out.append(paper_conjuntos[line]['id'])
        a=paper_conjuntos[line]['references']
        for tag in range(len(a)):
            try:
                b=ides_in[a[tag]]
                b+=1
                ides_in.update({a[tag]:b})
            except:
                ides_in[a[tag]]=1
    return(ides_in)

#Crea el archivo de papers filtrados, la lista y el diccionario de frecuencias de citas
def agregar_contar(ides_in,ides_out,R_n,paper_conjuntos):
    paper_conjuntos=filtrar_papers(R_n,'Nature')
    ides_in=contador(ides_in,ides_out,paper_conjuntos)
    return(ides_in,ides_out)
    
ides_out=[]
ides_in={}
for i in range(160,167):
    ides_in,ides_out=agregar_contar(ides_in,ides_out,str(i),[])
    print(i)
    
#ides_in,ides_out=agregar_contar(ides_in,ides_out,str(162),[])

606
160
651
161
640
162
630
163
617
164
625
165
129
166


#### Esta es la rutina que hay que correr sobre los 166 archivos. En promedio va a dar unos 100.000 papers en total, antes de la intersección.

In [ ]:
papers_filtrados = []
version=1
for i in range(160,167):
    filtrar_papers(str(i),'Nature',version)
    print(i)
    